### DISCUSSION TITLE
The path for rulesetDirectories is related to which directory?

### DISCUSSION THREAD

**Discussion by Snailya** [iTwin/itwinjs-core#3511] [Q&A] [answered]
I want to use presentation ruleset by id in my widget, but it seems that no ruleset was found in PresentationManager.

The rulesetDirectoryies is defined in backend's main.ts as follow. Print ruleset directly after initialized, _registeredRulesets property is a map of size 0.
![image](https://user-images.githubusercontent.com/19658624/163117688-0252f8a2-3627-4976-ac0f-f60bc092d504.png)
![image](https://user-images.githubusercontent.com/19658624/163117802-a1bd3e49-08d0-4e9d-93d9-e9b350c8be8b.png)

Here's my project folder structure, am I missing some steps, or the rulesetDirectories path are incorrect?
![image](https://user-images.githubusercontent.com/19658624/163118003-6085a7b4-3e3b-4a7d-9a9e-dee7ccd9f4b2.png)

**Comment by grigasp** [comment, marked as answer]
The relative paths set in `rulesetDirectories` attribute are resolved based on `process.cwd()`. You can also pass a full path to be sure. 

Also, when looking for rulesets by ID, the library only looks at files with `.PresentationRuleSet.json` extension. 

I see both of these points aren't documented well enough, I'll make sure to improve that. Thanks.

**Reply by Snailya** [reply]
I rename my ruleset files with `.PrensentationRuleSet.json` extension, and set paths with absolute path. It seem that still something I missed as no registeredRulesets.
![image](https://user-images.githubusercontent.com/19658624/163295702-cc8718ec-bca4-4fa9-bf0f-03ed953f2c65.png)
![image](https://user-images.githubusercontent.com/19658624/163295899-8981bf06-7a79-482f-b0e3-2d7397d405e0.png)
![image](https://user-images.githubusercontent.com/19658624/163295914-3170c28f-5fad-408a-9016-a6fa3ea597b2.png)

**Reply by grigasp** [reply]
They're not going to appear in that map - the file-based rulesets are read and cached in the native node addon. They're going to be read when you issue a presentation request with a ruleset id.

**Reply by Snailya** [reply]
Thanks. 
It works when issue a presentation request with ruleset id.
